edgedbpkg/postgresql/patches/postgresql-edgedb__gate_statement_log-14.patch

```diff
@@ -1,28 +1,27 @@
 From 33b7d8a7aa52caa5425df91156ee2d64523f5062 Mon Sep 17 00:00:00 2001
 From: Fantix King <fantix.king@gmail.com>
 Date: Fri, 30 Aug 2024 13:34:32 -0400
-Subject: [PATCH 6/6] Gate long STATEMENT log in SERIALIZATION error
+Subject: [PATCH 6/6] Gate long STATEMENT log
 
 ---
  src/backend/utils/error/elog.c | 5 ++++-
- 1 file changed, 4 insertions(+), 1 deletion(-)
+ 1 file changed, 3 insertions(+), 1 deletion(-)
 
 diff --git a/src/backend/utils/error/elog.c b/src/backend/utils/error/elog.c
 index 8d33686510a..bd3197606d0 100644
 --- a/src/backend/utils/error/elog.c
 +++ b/src/backend/utils/error/elog.c
-@@ -3106,7 +3106,10 @@ send_message_to_server_log(ErrorData *edata)
+@@ -3106,7 +3106,9 @@ send_message_to_server_log(ErrorData *edata)
  	 */
  	if (is_log_level_output(edata->elevel, log_min_error_statement) &&
  		debug_query_string != NULL &&
 -		!edata->hide_stmt)
 +		!edata->hide_stmt &&
-+		// Hide STATEMENT in serialization error if longer than 128 bytes
-+		(edata->sqlerrcode != ERRCODE_T_R_SERIALIZATION_FAILURE ||
-+			strnlen(debug_query_string, 128) < 128))
++		// Hide STATEMENT if longer than 1024 bytes
```

Comment:
Can we truncate instead of omitting the statement altogether?

Reply:
Yeah, fixed!